chromadb/api/types.py

```diff
@@ -167,7 +169,7 @@ class IncludeEnum(str, Enum):
 
 class GetResult(TypedDict):
     ids: List[ID]
-    embeddings: Optional[List[Embedding]]
+    embeddings: Optional[Union[List[Embedding], List[PyEmbedding], NDArray]]
```

Comment:
this is probably intentional but it's not immediately clear to me why `NDArray` is part of the union here (and other places below)

Reply:
We return the `GetResult`/`QueryResult` directly to the user. Although our internal representation of embeddings is a python list of numpy arrays, we want to return a 2D numpy array to the user. Thus, we need to convert before returning and the structure needs to support an `NDArray` for this.

Reply:
sorry, a little confused; we accept embeddings from the user as `List[NDArray]` but return them as a 2D `NDArray`?

Reply:
We accept as either `List[List]` or `NDArray` and return as `NDArray`. `List[NDArray]` is only used for our internal representation.